src/llm/apis/openai_completions.cpp

```diff
@@ -75,32 +76,139 @@ absl::Status OpenAIChatCompletionsHandler::parseCompletionsPart() {
+                        auto entry = v.GetObject();
+                        if (!entry.HasMember("type") || !entry["type"].IsString()) {
+                            return absl::InvalidArgumentError("Invalid message structure - content object type missing");
+                        }
+                        auto type = entry["type"].GetString();
+                        if (type == std::string("text")) {
+                            if (!entry.HasMember("text") || !entry["text"].IsString()) {
+                                return absl::InvalidArgumentError("Invalid message structure - content text missing");
+                            }
+                            contentText = entry["text"];
+                            continue;
+                        } else if (type == std::string("image_url")) {
+                            if (!entry.HasMember("image_url") || !entry["image_url"].IsObject()) {
+                                return absl::InvalidArgumentError("Invalid message structure - content image_url missing");
+                            }
+                            auto imageUrl = entry["image_url"].GetObject();
+                            if (!imageUrl.HasMember("url") || !imageUrl["url"].IsString()) {
+                                return absl::InvalidArgumentError("Invalid message structure - image_url does not have url field");
+                            }
+                            std::string url = imageUrl["url"].GetString();
+                            std::string pattern = ";base64,";
+                            std::size_t found = url.find(pattern);
+                            size_t offset = found + pattern.length();
+                            if (found == std::string::npos) {
+                                offset = 0;
+                            }
+                            url.erase(0, offset);
+                            std::string decoded;
+                            if (!absl::Base64Unescape(url, &decoded)) {
+                                return absl::InvalidArgumentError("Invalid base64 string in request");
+                            }
+                            std::vector<unsigned char> data(decoded.begin(), decoded.end());
+                            cv::Mat dataMat(data, true);
+                            cv::Mat image;
+                            try {
+                                image = cv::imdecode(dataMat, cv::IMREAD_UNCHANGED);
+                            } catch (const cv::Exception& e) {
+                                return absl::InvalidArgumentError("Error during string to mat conversion");
+                            }
+                            std::vector<size_t> shape;
+                            shape.push_back(image.rows);
+                            shape.push_back(image.cols);
+                            shape.push_back(image.channels());
+                            ov::Tensor tensor(getOvTypeFromMatType(image.depth()), shape);
```

Comment:
shouldnt we check if getOvTypeFromMatType is undefined and reject such image? do we have test for that?